Update audit-logging.adoc (#254)
mulesoft/docs-access-management: modules/ROOT/pages/audit-logging.adoc

https://anypoint.mulesoft.com/exchange/portals/anypoint-platform/f1e97bc6-315a-4490-82a7-23abe036327a.anypoint-platform/audit-log-query-api/[Audit Log REST API] from the Audit Logging Query API and its
https://anypoint.mulesoft.com/exchange/portals/anypoint-platform/organizations/8bfc8bbf-5508-419e-aadc-77dfe18a8172/assets/f1e97bc6-315a-4490-82a7-23abe036327a.anypoint-platform/audit-log-query-api/1.0.5/files/fat-raml/zip[RAML].

=== Use a Query Loop to make Requests in Audit Logs

> To query audit logs, the current https://anypoint.mulesoft.com/exchange/portals/anypoint-platform/f1e97bc6-315a-4490-82a7-23abe036327a.anypoint-platform/audit-log-query-api/minor/1.0/console/method/%23245/ [query api] will continue to work.

Author: I'm not sure what you're trying to communicate here. Is this the old API?

Reviewer: Yes this is the old api and we are not retiring them yet so they are still valid

Author: ```suggestion
The https://anypoint.mulesoft.com/exchange/portals/anypoint-platform/organizations/8bfc8bbf-5508-419e-aadc-77dfe18a8172/assets/f1e97bc6-315a-4490-82a7-23abe036327a.anypoint-platform/audit-log-query-api/1.0.5/files/fat-raml/zip[latest version of the query API] uses query pagination for efficiency, but you can continue to use the https://anypoint.mulesoft.com/exchange/portals/anypoint-platform/f1e97bc6-315a-4490-82a7-23abe036327a.anypoint-platform/audit-log-query-api/minor/1.0/console/method/%23245/[previous version of the query API], which uses offset pagination and returns the total number of records.
```

> Pagination in that api is via offset in the query payload, and the response returns the total number of records.

Author: Same with this. I am not really sure what this means.

Reviewer: The old api uses offset pagination.
The new api format user query pagination.
So calling out the old method of pagination so we can explain that the new method uses a different pagination.

Author: ```suggestion
```

Author: Proposing removing this line and consolidating this information into the above paragraph. Let me know what you think.

> We are adding a new pagination using cursor and you can initiate query loops by using cursors. This cursor paging can be used by using the `cursorPagination` query parameter and setting it to true.

Author: ```suggestion
The newer version of the API uses pagination with cursors, so you can initiate query loops using cursors. Setting the `cursorPagination` query parameter to true enables you to use cursor paging.
```

Reviewer: Yes this looks good.

> When using cursor paging, each response will return a cursor. This cursor corresponds to the last data entry in the response data set. So the next request in query loop should include the cursor value that was returned in the previous response, and so on.

Author: ```suggestion
When you use cursor paging, each response returns a cursor. The cursor corresponds to the last data entry in the response data set. The subsequent request in the query loop should use the cursor value that was returned in the previous response, and so on.
```

Reviewer: When you use cursor paging, each response returns a cursor. The cursor corresponds to the last data entry in the response data set. 

The subsequent request in the query loop should use the cursor value that was returned in the previous response, and so on.

Reviewer: We send the response and so when user makes a subsequent request, they should use the cursor that we sent previously.

> When the data set is empty in response, then you have reached the end of query loop, and there is no more data to query for that time window. So, that is the indicator the stop the query loop. When the data set is empty in the response, there will be no cursor returned.

Author: ```suggestion
When the data set is empty in response, you have reached the end of the query loop, and there is no more data to query for that time window. At that point, stop the query loop. When the data set is empty in the response, no cursor is returned.
```

Reviewer: This looks good.

> Total is not returned by default when using cursor pagination, as the query loop is not dependent on offset and number of total rows.

Author: ```suggestion
When you use cursor pagination, `total` is not returned by default, because the query loop is not dependent on the offset or the number of total rows.
```

> When using cursor paging, you can opt in to getting total by using the `includeTotal` query parameter and setting it to true.

Author: ```suggestion
If you want to see the total when you use cursor paging, set the `includeTotal` query parameter to true.
```

> When using `cursorPagination` query param as true, if you send `offset` and `cursor` in the query body, we’ll give preference to cursor and ignore the offset.

Author: ```suggestion
When the `cursorPagination` query parameter is true and you use`offset` and `cursor` in the query body, `cursor` is prioritized, and `offset` is ignored.
```

There is no mapping between `offset` and `cursor` values; And so when you run a query loop over a time window, you cannot switch to the new param somewhere in the middle by using the `cursorPagination` and continue the loop. This is because when using `cursorPagination`, `offset` is ignored and so the loop will be start from the beginning.

The following example shows a series of calls in a query loop for cursor based query.

> . The first call does not have a cursor in the body; And it does not request for total. The response returns the data and a: 

Author: ```suggestion
. The first call does not have a cursor in the body and does not request the `total`. The response returns the following data: 
```

The newer version of the API uses pagination with cursors, so you can initiate query loops using cursors. Setting the `cursorPagination` query parameter to true enables you to use cursor paging.
When you use cursor paging, each response returns a cursor. The cursor corresponds to the last data entry in the response data set. The subsequent request in the query loop should use the cursor value that was returned in the previous response, and so on.
When the data set is empty in response, you have reached the end of the query loop, and there is no more data to query for that time window. At that point, stop the query loop. When the data set is empty in the response, no cursor is returned.
When you use cursor pagination, `total` is not returned by default, because the query loop is not dependent on the offset or the number of total rows.
If you want to see the total when you use cursor paging, set the `doIncludeTotal` query parameter to true.

> When the `cursorPagination` query parameter is true and you use`offset` and `cursor` in the query body, `cursor` is prioritized, and `offset` is ignored.

Author: ```suggestion
When the `cursorPagination` query parameter is true and you use `offset` and `cursor` in the query body, `cursor` is prioritized, and `offset` is ignored.
```

There is no mapping between `offset` and `cursor` values; And so when you run a query loop over a time window, you cannot switch to the new param somewhere in the middle by using the `cursorPagination` and continue the loop. This is because when using `cursorPagination`, `offset` is ignored and so the loop will be start from the beginning.

> This cursor pagination api is not applicable to the csv query, and that continues to use the offset based pagination.

Reviewer: @sarathecoubian 
I added this line 'This cursor pagination api is not applicable to the csv query, and that continues to use the offset based pagination.'
The csv based api doesn't use the cursor based pagination and it'll continue to use the old offset based paging api.

Author: ```suggestion
The cursor-based pagination API is not applicable to the CSV query. The query continues to use offset-based pagination.
```

	}
----

> . The third call is the final call in a loop. There are no more logs to be returned, and so the response contains no data.  Response doesn't have a cursor too as there are no more logs to be returned.

Author: ```suggestion
. The third call is the final call in a loop. Because there are no more logs to be returned, the response does not have a cursor and contains no data.
```
